Mathlib/NumberTheory/ModularForms/EisensteinSeries/QExpansion.lean

```diff
@@ -181,3 +192,121 @@ theorem EisensteinSeries.qExpansion_identity_pnat {k : ℕ} (hk : 1 ≤ k) (z :
   · simp [show k ≠ 0 by grind]
   · apply (summable_pow_mul_cexp k 1 z).congr
     simp
+
+lemma summable_prod_eisSummand {k : ℕ} (hk : 3 ≤ k) (z : ℍ) :
+    Summable fun x : ℤ × ℤ ↦ eisSummand k ![x.1, x.2] z := by
+  rw [← summable_norm_iff, ← (piFinTwoEquiv fun _ ↦ ℤ).summable_iff, piFinTwoEquiv_apply]
+  apply (summable_norm_eisSummand (by linarith) z).congr
+  simp [eisSummand]
+
+lemma tsum_eisSummand_eq_tsum_sigma_cexp {k : ℕ} (hk : 3 ≤ k) (hk2 : Even k) (z : ℍ) :
+    ∑' v, eisSummand k v z = 2 * riemannZeta k + 2 * ((-2 * π * I) ^ k / (k - 1)!) *
+    ∑' (n : ℕ+), σ (k - 1) n * cexp (2 * π * I * z) ^ (n : ℕ) := by
+  rw [← (piFinTwoEquiv fun _ ↦ ℤ).symm.tsum_eq, Summable.tsum_prod
+    (by apply summable_prod_eisSummand hk z), tsum_int_eq_zero_add_two_mul_tsum_pnat]
+  · have (b : ℕ+) := qExpansion_identity_pnat (k := k - 1) (by grind)
```

Comment:
This is a bit confusing (it means something different from `have b : ℕ+ := ...`), so better give it a name.